base/applications/osk/settings.c

```diff
@@ -58,14 +57,14 @@ BOOL LoadDataFromRegistry()
     }
 
     /* Load the data value (it can be either FALSE or TRUE depending on the data itself) */
-    Globals.bShowWarning = (dwShowWarningData != 0);
+    Globals.bShowWarning = (dwValue != 0);
 
     /* Query the key */
     lResult = RegQueryValueExW(hKey,
                                L"IsEnhancedKeyboard",
                                0,
                                0,
-                               (BYTE *)&dwLayout,
+                               (BYTE *)&dwValue,
                                &cbData);
 
     if (lResult != ERROR_SUCCESS)
```

Comment:
This is not a bug introduced by this commit, but the check should be improved to also validate that `cbData == sizeof(dwValue)` (Something for a next PR maybe?)

Reply:
You're right, thank you for your idea! I'll address this in the next PR.